plugins/admission/pkg/opa/opa.go

```diff
@@ -0,0 +1,88 @@
+package opa
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/fairwindsops/insights-plugins/opa/pkg/opa"
+	"github.com/thoas/go-funk"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/fairwindsops/insights-plugins/admission/pkg/models"
+)
+
+const opaVersion = "0.2.8"
+
+// ProcessOPA runs all checks against the provided Custom Check
+func ProcessOPA(ctx context.Context, obj map[string]interface{}, resourceName, apiGroup, resourceKind, resourceNamespace string, configuration models.Configuration) (models.ReportInfo, error) {
+	report := models.ReportInfo{
+		Report:  "opa",
+		Version: opaVersion,
+	}
+	actionItems := make([]opa.ActionItem, 0)
+	for _, instanceObject := range configuration.OPA.CustomCheckInstances {
+		instance := opa.CustomCheckInstance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: instanceObject.AdditionalData.Name,
+			},
+			Spec: opa.CustomCheckInstanceSpec{
+				CustomCheckName: instanceObject.CheckName,
+				Output:          instanceObject.AdditionalData.Output,
+				Parameters:      instanceObject.AdditionalData.Parameters,
+				Targets: funk.Map(instanceObject.Targets, func(s string) opa.KubeTarget {
+					splitValues := strings.Split(s, "/")
+					return opa.KubeTarget{
+						APIGroups: []string{splitValues[0]},
+						Kinds:     []string{splitValues[1]},
+					}
+				}).([]opa.KubeTarget),
+			},
+		}
+		found := false
+		for _, target := range instance.Spec.Targets {
+			if apiGroup == target.APIGroups[0] && resourceKind == target.Kinds[0] {
```

Comment:
Should we worry about there being more than one apigroup/kind?

Reply:
Ideally, in this case though those arrays were just created by a request to Insights that returns an array of strings where it's `group/kind` and then splitting them. So in this particular case until/if we update Insights to store an array of groups and an array of kinds this is all we'll get back.